sharedbuffers/mapped_struct.py

```diff
@@ -324,13 +331,17 @@ class mapped_dict(dict):
 
     @classmethod
     def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
-        return mapped_list.pack_into((obj.keys(), [obj[k] for k in obj]), buf, offs, idmap, implicit_offs)
+        keys = list(obj)
+        values = [obj[k] for k in keys]
+        return mapped_list.pack_into([keys, values], buf, offs, idmap, implicit_offs)
 
     @classmethod
     def unpack_from(cls, buf, offs, idmap = None):
         key, values = mapped_list.unpack_from(buf, offs, idmap)
         return cls(zip(key, values))
 
+@cython.ccall
+@cython.returns(cython.char)
```

Comment:
Why not int?

Also, this masks errors. Better add the declaration to the pxd and use the except syntax to specify an error value, so that it can throw exceptions.

Reply:
Why not char if the returns values are -1, 0 and 1?

Reply:
In my experience, you don't gain anything from using a narrower return type. In fact, the ABI is more cumbersome and slower for narrower types, `int` is C's native type and it tends to outperform narrower types. So unless there's a type congruence reason to use char (say, you want to match the type of some other API) or space reasons to narrow it down (say, it's going into an array or struct), you're better off using int.